src/hooks/calls/user/useAuthentication.js

```diff
@@ -0,0 +1,80 @@
+import { useEffect } from 'react';
+import useSWRImmutable from 'swr/immutable'
+import { useUserApi } from "../../api/useUserApi";
+
+export function useAuthentication({tokenName, username, password, userClient, options, configuration, axios}) {
+  const _userClient = useUserApi({ userClient, username, password, ...configuration, axios});
+
+  //Fetch function to get token and delete old ones
+  const fetchAuth = async (username) => {
+
+    if (!username || !password || !tokenName) throw new Error("missing tokenName, username or password");
+
+    const allTokens = await _userClient
+      .userGetTokens(username)
+      .then(({ data }) => data);
+    const appTokens = allTokens?.filter((item) => item.name === tokenName);
+
+    if (appTokens?.length > 0) {
+      appTokens.forEach((token) => {
+        _userClient
```

Comment:
Is an "await" needed here?

Reply:
That script is for deleting old tokens, we are actually not using the response from those requests, and if that request fails it won't brake anything, so I think we shouldn't wait for that to finish.